json.c

```diff
@@ -41,18 +41,35 @@ const struct _json_value json_value_none;
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
-#include <math.h>
 
 typedef unsigned int json_uchar;
 
-/* There has to be a better way to do this */
-static const json_int_t JSON_INT_MAX = sizeof(json_int_t) == 1
-                                       ? INT8_MAX
-                                       : (sizeof(json_int_t) == 2
-                                         ? INT16_MAX
-                                         : (sizeof(json_int_t) == 4
-                                           ? INT32_MAX
-                                           : INT64_MAX));
+#define JSON_INT_MAX ((1ULL << ((sizeof(json_int_t) * 8) - 1)) - 1)
```

Comment:
Shouldn't this be using [`CHAR_BIT`](https://en.cppreference.com/w/cpp/types/climits) instead of `8`?